src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -0,0 +1,369 @@
+            type="button"
+            title="Reattach volume"
+            onClick={() => {
+              ensureEditMode(formik);
+              removeDevice(noneDeviceId, formik);
+            }}
+            className="has-icon u-no-margin--bottom"
+          >
+            <Icon name="connected"></Icon>
+            <span>Reattach</span>
+          </Button>
+        ) : (
+          <Button
+            appearance="base"
+            type="button"
+            onClick={() => {
+              ensureEditMode(formik);
+              addNoneDevice(item.key, formik);
+            }}
+            className="has-icon u-no-margin--bottom"
+            dense
+          >
+            <Icon name="disconnect"></Icon>
+            <span>Detach</span>
+          </Button>
+        ),
+      }),
+    );
+
+    Object.keys(item.proxy).forEach((key) => {
+      if (key === "name" || key === "type") {
+        return null;
+      }
+
+      inheritedRows.push(
+        getInheritedDeviceRow({
+          label: deviceKeyToLabel(key),
+          inheritValue: item.proxy[key as keyof typeof item.proxy],
+          readOnly: false,
+          isDeactivated: isNoneDevice,
+        }),
+      );
+    });
+  });
+
+  const customRows: MainTableRow[] = [];
+  formik.values.devices.forEach((formDevice, index) => {
+    if (formDevice.type !== "proxy") {
+      return;
+    }
+    const device = formik.values.devices[index] as LxdProxyDevice;
+
+    customRows.push(
+      getConfigurationRowBase({
+        className: "no-border-top custom-device-name",
+        configuration: (
+          <RenameDeviceInput
+            name={device.name}
+            index={index}
+            setName={(name) => {
+              ensureEditMode(formik);
+              void formik.setFieldValue(`devices.${index}.name`, name);
+            }}
+          />
+        ),
+        inherited: "",
+        override: (
+          <Button
+            className="u-no-margin--top u-no-margin--bottom"
+            onClick={() => {
+              ensureEditMode(formik);
+              removeDevice(index, formik);
+            }}
+            type="button"
+            appearance="base"
+            hasIcon
+            dense
+            title="Detach Proxy"
+          >
+            <Icon name="disconnect" />
+            <span>Detach</span>
+          </Button>
+        ),
+      }),
+    );
+    customRows.push(
+      getProxyDeviceFormRows("NAT mode", "nat", index, [
+        { label: "Enabled", value: "true" },
+        { label: "Disabled", value: "false" },
+      ]),
+    );
+    customRows.push(
+      getProxyDeviceFormRows("Bind", "bind", index, [
+        { label: "Host", value: "host" },
+        { label: "Instance", value: "instance" },
+      ]),
+    );
+    customRows.push(
+      getProxyDeviceFormRows("Use HAProxy Protocol", "proxy_protocol", index, [
+        { label: "Yes", value: "true" },
+        { label: "No", value: "false" },
+      ]),
+    );
```

Comment:
```suggestion
      getProxyDeviceFormRows("Use HAProxy Protocol", "proxy_protocol", index, optionYesNo, device.proxy_protocol),
```
We have most of the yes/no, enabled/disabled, etc options defined as constants to reuse.

We can add another parameter to the `getProxyDeviceFormRows` to pass in the current value.

Reply:
Why do we need to pass through the current value?

Reply:
We don't have to, but I think it is easiest to do it this way, because otherwise we have to pick the device at index again and cast it to the right typescript type. Passing the value seems a bit simpler.

Reply:
Okay, but what attribute within the Select component should accept the value? I am not sure what is supposed to be done with it.

Reply:
On the select it can be `value={value ?? ""}`

Reply:
Ah I see, thank you.